src/VisualStudio/Core/Def/ProjectSystem/MetadataReferences/VisualStudioMetadataReferenceManager.MetadataCache.cs

```diff
@@ -13,15 +13,10 @@ internal sealed partial class VisualStudioMetadataReferenceManager
     {
         private sealed class MetadataCache
         {
-            private const int InitialCapacity = 64;
-            private const int CapacityMultiplier = 2;
-
             private readonly object _gate = new();
 
             // value is ValueSource so that how metadata is re-acquired back are different per entry. 
-            private readonly Dictionary<FileKey, ValueSource<AssemblyMetadata>> _metadataCache = new();
-
-            private int _capacity = InitialCapacity;
+            private readonly Dictionary<FileKey, AssemblyMetadata> _metadataCache = new();
```

Comment:
because all values in this dict were always COnstantValueSources, we can just elide the ValueSource part and point directly at the assemblymetadata itself.  this simplifies a lot of hte code/signatures here.